### PR TITLE
Unmute SimpleThreadPoolIT.testThreadPoolMetrics with additional logging

### PR DESCRIPTION
**Pull request by mosche**
Unmute `SimpleThreadPoolIT.testThreadPoolMetrics` with additional logging and use a more vocal assertion to better understand test failures (relates to #104652)

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java [hunk 1]
```diff
@@ -147,30 +154,36 @@ public void testThreadPoolMetrics() throws Exception {
...
-            );
-            threadPoolMetrics.forEach((suffix, value) -> {
-                String metricName = ThreadPool.THREAD_POOL_METRIC_PREFIX + stats.name() + suffix;
-                List<Measurement> measurements;
-                if (suffix.equals(ThreadPool.THREAD_POOL_METRIC_NAME_COMPLETED)) {
-                    measurements = plugin.getLongAsyncCounterMeasurement(metricName);
-                } else {
-                    measurements = plugin.getLongGaugeMeasurement(metricName);
-                }
+            Map<String, Long> threadPoolStats = List.of(
```
Comment (ldematte) [inline server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java:159]: Nit: why not Map.ofEntries directly?

File: server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java [hunk 2]
```diff
@@ -111,7 +118,7 @@ public void testThreadNames() throws Exception {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/104652")
+    // temporarily re-enable to gather
```
Comment (ldematte) [inline server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java:121]: Looks like this comment is unfinished

**Review by ldematte — APPROVED** [review]
OK, left a couple of very minor comments

File: server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java (hunk 1 above)
Comment (mosche) [reply server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java:159]: This creates a sorted map. That's just for convenience to have a consistent order when logging the expect and actual result further below.

Comment (ldematte) [reply server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java:159]: Oh right, my eyes skipped over the "sorted" :) makes sense!

**Comment by mosche** [comment]
Fixed in https://github.com/elastic/elasticsearch/pull/106580

**Pull request merged**